Add prefix and architectures for publish; various small fixes

mxey opened the pull request:
I have used your module to set up our mirror with Aptly and run into some issues and missing features.

I made this a single pull request, since the commits are based on each other, but if you would rather I split them up, I can do that too (would be some work though)

mxey committed:
Add support for publish prefixes

Also fixes exists? check of publishes

mxey committed:
Avoid slow Puppet file recursion on the whole package pool

mxey committed:
Avoid permission issues by switching to world-readable directory when running Aptly

mxey committed:
Remove error suppression when aptly commands fail

Lead to hard to debug issues if these Aptly commands failed because Puppet tried
to create resources that already existed

mxey committed:
Add configurable architectures to aptly::publish

According to Aptly warning messages, architectures should be specified
if publishing an empty repository, because the list of architectures
cannot be changed after the fact

mxey committed:
Add label support to aptly::publish

mxey committed:
Ensure Aptly is installed and configured before using it

aerostitch commented:
Hi @mxey ,

I don't have merge rights on the repo anymore but there are some complains from rubocop that would be easy to fix here: https://travis-ci.org/tubemogul/puppet-aptly/jobs/265191271

It also seems that if they still want to support ruby 1.9 you would have to pin the public_suffix gem to version `< 1.5.0` if ruby `< 2.0` in the Gemfile.

That should make your tests pass (https://travis-ci.org/tubemogul/puppet-aptly) and probably help on the PR integration.

Joseph

mxey committed:
Add notice to CONTRIBUTING.md about Rubocop

mxey commented:
I fixed the RuboCop warnings and added a mention of RuboCop to CONTRIBUTING.md so the next person will know.

Unfortunately, I cannot reproduce the issue with public_suffix locally. That gem does not even show up in my bundle, let alone fail to install, neither on macOS nor in a Ubuntu 12.04 Docker container (Travis CI uses Ubuntu 12.04)

mxey committed:
Fix build by updating Bundler 

Fix build failure Gem::InstallError: public_suffix requires Ruby version >= 2.1.

mxey committed:
Fix build by pinning old version of metadata-json-lint for Ruby < 2

fionera committed:
feat: Add Support for mirror creation with package filter

Commit:
fix: Delete undef parameters in cli
